### DISCUSSION TITLE
Make titles in bibliography entries hyperlinks

### DISCUSSION THREAD

**Discussion by alexanderkoller** [typst/typst#4420] [Q&A] [answered]
Hi,

I just discovered Typst and am really enthusiastic about it, but here's one thing I can't figure out. In my scientific community, we use a bibliography style in which URL or DOI fields in the Bibtex entry are converted into hyperlinks on the paper title. See [the PDF on this web page](https://aclanthology.org/2020.acl-main.463/) or the image below for an example.

Is there any way to achieve this behavior in Typst? Things I have already tried and that didn't work:

- Generate a `#link` call in the CSL; this is simply rendered verbatim, without executing the `link` function.
- Generate some text of the form `%title%url%` in the CSL and use a regex show rule to convert it into a hyperlink; but for some reason, I can't get a regular expression to match a colon, and of course the `//` in the URL will be interpreted as starting a comment.

I love Typst and I love my community's bibliography style - can I have both at the same time?

![Screenshot 2024-06-19 at 22 11 43](https://github.com/typst/typst/assets/8291392/dcbe0d9a-3d22-4ca4-bcd0-f188abf89630)

**Comment by jamesrswift** [comment]
I definitely agree. I did some small investigating last night and I managed to both a solution for DOI links, shown below:

```typ
#show link: set text(fill: blue)
#show link: underline

#show regex("(\d+\.\d+/.*)"): (it)=>{
  link("http://doi.org/"+it.text, it)
}
```

This works because regex doesn't match across different text elements (which the different fields of the bibliography are).

I think a more idiomatic solution would be along the lines of:

```
#show bibliography.doi:  (it) => {link("http://doi.org/"+it.text, it)}
```

Similar show rules could be useful for url fields, and probably many others (I'm thinking ORCID links)

**Reply by alexanderkoller** [reply]
Thank you for the suggestion!

This makes the DOI field a hyperlink, but it does not achieve the effect I showed in my original post, where the DOI itself is not even shown in the rendered document, but the paper title itself is made a hyperlink to the DOI/URL. Do you have thoughts on how to achieve this?

For another idea, is it possible to import the [biblatex](https://github.com/typst/biblatex) crate into a Typst document and simply write some code to generate the bib entries, rather than going through CSL?

**Comment by alexanderkoller** [comment, marked as answer]
Just to close my own issue, this can now be done using my [blinky](https://typst.app/universe/package/blinky/) package.
